scripts/create_assignments_from_csv.py

```diff
@@ -115,17 +118,25 @@ def main(arguments):
     for worker in workers:
         workers_dict[worker.user_id] = worker
 
+    if not (args.x_field and args.y_field):
+        addresses = batch_geocode(locations, out_sr=args.wkid)
     assignments_to_add = []
+    i = 0
     for assignment in assignments_in_csv:
         assignment_to_add = workforce.Assignment(project,
                                                  assignment_type=assignment_type_dict[assignment[args.assignment_type_field]],
                                                  )
 
         # Create the geometry
-        geometry = dict(x=float(assignment[args.x_field]),
+        if args.x_field and args.y_field:
+            geometry = dict(x=float(assignment[args.x_field]),
                         y=float(assignment[args.y_field]),
                         spatialReference=dict(
                             wkid=int(args.wkid)))
+        else:
+            location_geometry = addresses[i]['location']
```

Comment:
what happens if no location is returned? (e.g. an invalid address)

Reply:
a dictionary is returned with no location field - so i have updated to handle what will likely be a KeyError